uarray/small_dynamic_array.h

```diff
@@ -0,0 +1,228 @@
+#pragma once
```

Comment:
We should probably add a copyright note if this is taken from somewhere.

Reply:
I wrote it.